pkg/app/piped/deploysource/deploysource.go

```diff
@@ -171,16 +185,40 @@ func (p *provider) prepare(ctx context.Context, lw io.Writer) (*DeploySource, er
 	}
 	fmt.Fprintln(lw, "Successfully loaded the application configuration file")
 
+	var (
+		embedded     embedded
+		embeddedType = EmbeddedTypeNone
+	)
 	// Decrypt the sealed secrets if needed.
 	if gac.Encryption != nil && p.secretDecrypter != nil && len(gac.Encryption.DecryptionTargets) > 0 {
+		embedded.secret = true
+	}
+
+	if gac.Attachment != nil && len(gac.Attachment.Targets) > 0 {
+		embedded.attachment = true
+	}
+
+	if embedded.secret && embedded.attachment {
+		embeddedType = EmbeddedTypeBoth
+	} else if embedded.secret {
+		embeddedType = EmbeddedTypeSecretOnly
+	} else if embedded.attachment {
+		embeddedType = EmbeddedTypeAttachmentOnly
+	}
```

Comment:
I understand this approach, but can we have some better for future embedding combinations? I mean, currently, we have attachment and secretEncrypter so that we can have if/else with BOTH flag, but if I add one more processor, this could be too complicated to deal with wdyt?

Reply:
In fact, I have implemented an approach that supports what I mentioned above, but it still needs to be finished. You can check it here 👀
My implementation has a problem with `only run go template once`, so it can't perform encrypt->attach 😓